core/src/main/java/nu/yona/server/exceptions/ResourceBasedException.java

```diff
@@ -1,21 +1,44 @@
 package nu.yona.server.exceptions;
 
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.event.ContextRefreshedEvent;
+import org.springframework.context.event.EventListener;
 import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Component;
+
+import nu.yona.server.Translator;
 
 /**
  * This exception class is used as a base for all exceptions that use an error message which is defined in the message properties
- * 
- * @author pgussow
  */
 public abstract class ResourceBasedException extends RuntimeException
 {
+	@Component
+	private static class TranslatedTextRetriever
+	{
+		@Autowired
+		private Translator translator;
+
+		@EventListener({ ContextRefreshedEvent.class })

```

Comment:
But the problem that this class solves is that the translator cannot be injected by autowiring. So you could choose to set a `private static Translator translator;` (instead of `private static TranslatedTextRetriever translatedTextRetriever;`) in `ResourceBasedException` and rename `TranslatedTextRetriever` to `TranslatorInjector` and remove the `getLocalizedText` method from it. It would be less code and less 'magic' to me.